### PR TITLE
Transfer Assets - Part 1 (Support to RgbWallet)

### PR DESCRIPTION
**Pull request by oacdutra**
**Decription**

This PR intent add initial support to RgbWallet, the structure to manage UTXOs and Tapret Commitments. This is a alternative to Amount Protocol.

Also, in the part 2, I will start implement all Asset Transfers tests.

**Tasks**
- [x] Support to RgbWallet (alternative to amount protocol).
   - [x] Retrieve commitments from PSBT
   - [x] Save commitments
   - [x] List UTXOs per contract
   - [x] List addresses used per contract
   - [x] List addresses used per contract
   - [x] Allow get next unused Address
   - [x] Allow get next unused UTXO
   - [x] Add esplora support
- [x] Integration Tests
  - [x] Setup docker to each integration tests
  - [x] Execute "send coins" directly to the node

**Other Tasks**
 - [x] Add import contract method

### CODE REVIEW

File: src/bin/bitmaskd.rs [hunk 1]
```diff
@@ -203,6 +217,7 @@ async fn main() -> Result<()> {
         .route("/contracts", get(contracts))
         .route("/interfaces", get(interfaces))
         .route("/schemas", get(schemas))
+        .route("/imports", post(imports))
```
Comment (cryptoquick) [inline src/bin/bitmaskd.rs]: Can we please change this to /import, singular, not plural?

File: src/rgb/imports.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+use amplify::{confinement::Confined, hex::FromHex};
```
Comment (cryptoquick) [inline src/rgb/imports.rs]: Change this filename from plural to singular ("import.rs")

File: src/rgb/resolvers.rs [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+use bitcoin::Script;
+use bitcoin_hashes::hex::FromHex;
 use bp::{LockTime, Outpoint, Tx, TxIn, TxOut, Txid, VarIntArray};
 use futures::executor;
+use rgb::{
+    prelude::{DeriveInfo, MiningStatus},
+    Utxo,
+};
 use rgbstd::resolvers::ResolveHeight;
 use rgbstd::validation::ResolveTx as ResolveCommiment;
```
Comment (cryptoquick) [inline src/rgb/resolvers.rs]:
```suggestion
use rgbstd::{resolvers::ResolveHeight, validation::ResolveTx as ResolveCommiment};
```

File: src/rgb/structs.rs [hunk 4]
```diff
@@ -1,5 +1,7 @@
 use bitcoin::Address;
-use std::str::FromStr;
+use bitcoin_scripts::address::AddressCompat;
+use rgb::{persistence::Stock, RgbWallet, TerminalPath};
+use std::{collections::HashMap, str::FromStr};
```
Comment (cryptoquick) [inline src/rgb/structs.rs]: Put stdlib imports at the top of the file, separate from the crates imports

File: src/structs.rs [hunk 5]
```diff
@@ -46,6 +46,7 @@ pub struct PublicWalletData {
     pub nostr_npub: String,
     pub xprvkh: String,
     pub xpubkh: String,
+    pub xpub: String,
```
Comment (cryptoquick) [inline src/structs.rs:49]: What is this xpub?

**Review by cryptoquick — CHANGES_REQUESTED** [review]
Very good work so far. Added a few review comments.

One question;

Can't we just use the same descriptors we use with BDK with RGB wallet also?

It's fine if we don't have an RGB integration test, the utils you build are really clever, but I still want those. They can be done in a separate PR, though.

File: src/structs.rs (hunk 5 above)
Comment (oacdutra) [reply src/structs.rs:49]: I will need plain xpub to build RgbWallet.

Comment (cryptoquick) [reply src/structs.rs:49]: Interesting... Good to know.

**Review by cryptoquick — APPROVED** [review]
All issues addressed. ACK.

**Comment by cryptoquick** [comment]
@josediegorobles Armando needs this merged, but let him know if you see any issues

**Pull request merged**